addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java

```diff
@@ -93,17 +93,54 @@ public void tearDown() {
     @Test
     public void testExtractText() throws Exception {
         PDFTextExtractor pdfte = new PDFTextExtractor(pdfFileBlob);
+        
+        // All text
         String text = pdfte.getAllExtractedLines();
         assertNotNull(text);
+                
         assertEquals(19045, text.length());
-        assertEquals("Contract Number: 123456789", pdfte.extractLineOf("Contract Number: "));
         assertFalse(text.contains("Something that is not in the file"));
         assertTrue(text.contains("13.1"));
+        
+        // Extract some text
+        assertEquals("Contract Number: 123456789", pdfte.extractLineOf("Contract Number: "));
         assertEquals("123456789", pdfte.extractLastPartOfLine("Contract Number: "));
     }
+    
+    @Test
+    public void testOperationExtractAlltext() throws Exception {
+        OperationChain chain = new OperationChain("testChain");
+        
+        ctx.setInput(testDoc);
+        chain.add(PDFExtractTextOperation.ID)
+             .set("save", false)
+             .set("targetxpath", "dc:description");
+        DocumentModel documentModified = (DocumentModel) automationService.run(ctx, chain);
+        String text = (String) documentModified.getPropertyValue("dc:description");
+        assertNotNull(text);
+        assertEquals(19045, text.length());
+        assertFalse(text.contains("Something that is not in the file"));
+        assertTrue(text.contains("13.1"));
+        
+        // Passing an empty patterntofind => all text
+        chain = new OperationChain("testChain");
+        ctx.setInput(testDoc);
+        chain.add(PDFExtractTextOperation.ID)
+             .set("save", false)
+             .set("targetxpath", "dc:description")
+             .set("patterntofind", "");
+        documentModified = (DocumentModel) automationService.run(ctx, chain);
+        text = (String) documentModified.getPropertyValue("dc:description");
+        assertNotNull(text);
+        assertEquals(19045, text.length());
+        assertFalse(text.contains("Something that is not in the file"));
+        assertTrue(text.contains("13.1"));
```

Comment:
You could factorize the 2 cases as the only difference is setting `patterntofind` or not.

Reply:
Ok, done in next commits